src/main/java/frc/robot/subsystems/vision/AprilTagVisionIOLimelight.java

```diff
@@ -48,57 +52,45 @@ public AprilTagVisionIOLimelight(String identifier) {
    */
   @Override
   public void updateInputs(AprilTagVisionIOInputs inputs) {
-    TimestampedString[] queue =
-        observationSubscriber.readQueue(); // Reads the queue of timestamped strings
     ArrayList<PoseEstimate> poseEstimates =
         new ArrayList<>(); // Creates an empty ArrayList to store pose estimates
 
-    // Iterates over each timestamped string in the queue
-    for (int i = 0; i < Math.min(queue.length, 3); i++) {
-      TimestampedString timestampedString = queue[i];
-      double timestamp = timestampedString.timestamp / 1e6; // Converts the timestamp to seconds
-      LimelightHelpers.Results results =
-          LimelightHelpers.parseJsonDump(timestampedString.value)
-              .targetingResults; // Parses the JSON dump and retrieves the targeting results
+    TimestampedDoubleArray[] queue =
+        new TimestampedDoubleArray[] {observationSubscriber.getAtomic()};
+
+    for (int i = 0; i < queue.length; i++) {
+      TimestampedDoubleArray timestampedDouble = queue[i];
+      double timestamp = timestampedDouble.timestamp / 1e6; // Converts the timestamp to seconds
+      double[] poseReading = timestampedDouble.value;
       Optional<Alliance> allianceOptional =
           DriverStation.getAlliance(); // Retrieves the alliance information from the DriverStation
 
+      // [tx,ty,tz,r,p,y,latency,tagcount, max tag span in meters, average tag distance in meters,
+      // average tag area]
       // Checks if there are no targets or if the alliance information is not present
-      if (results.targets_Fiducials.length == 0 || !allianceOptional.isPresent()) {
+      if (!allianceOptional.isPresent()) {
         continue; // Skips to the next iteration of the loop
       }
 
-      double latencyMS =
-          results.latency_capture
-              + results.latency_pipeline; // Calculates the total latency in milliseconds
+      double latencyMS = poseReading[6];
       Pose3d poseEstimation =
-          results.getBotPose3d_wpiBlue(); // Retrieves the pose estimation for the robot
-      double averageTagDistance = 0.0; // Initializes the average tag distance to 0.0
+          new Pose3d(
```

Comment:
probably shoudl ensure you're array is the correct size

Reply:
As long as there is something in the queue it will be.